.github/workflows/ci_rust.yml

```diff
@@ -0,0 +1,106 @@
+---
+name: Rust Bindings
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  # Pin the nightly toolchain to prevent breakage.
+  # This should be occasionally updated.
+  RUST_NIGHTLY_TOOLCHAIN: nightly-2022-04-05
+  MANIFEST_PATH: bindings/rust/Cargo.toml
+
+jobs:
+  generate:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: stable
+          override: true
+
+      - uses: camshaft/rust-cache@v1
+
+      - name: Generate
+        run: ./bindings/rust/generate.sh
+
+      - name: Test external build
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          cmake . -Bbuild -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=off
+          cmake --build build
+
+          export S2N_TLS_LIB_DIR=`pwd`/build/lib
+          export S2N_TLS_INCLUDE_DIR=`pwd`/api
+          export LD_LIBRARY_PATH=$S2N_TLS_LIB_DIR:$LD_LIBRARY_PATH
+
+          cd bindings/rust
+          ./generate.sh
+          ldd target/debug/integration | grep libs2n.so
+
+  rustfmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - uses: actions-rs/toolchain@v1.0.7
+        id: toolchain
+        with:
+          toolchain: ${{ env.RUST_NIGHTLY_TOOLCHAIN }}
+          profile: minimal
+          override: true
+          components: rustfmt
+
+      - uses: camshaft/rust-cache@v1
+
+      # We don't need to format the generated files,
```

Comment:
I'm wondering if we should start committing the generated files. We could then have a CI check to make sure they're up to date, similar to [s2n-quic's event check](https://github.com/aws/s2n-quic/blob/c5bb97fe426aba73f6836d898fe06d25f2760aab/.github/workflows/ci.yml#L530).

Reply:
To be fair, if we don't commit them, we don't need a CI job to make sure they're up to date.

It is a bit annoying having to call the generate script before anything else works. But idk if I'm totally convinced, and that wouldn't be part of this PR anyway ;)